src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DEFAULT_SIZE = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = DEFAULT_SIZE;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+        } else {
+            node.prev = tail;
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> element = findElement(index);
+            Node<T> node = new Node<>(element.prev, value, element);
+            if (element.prev == null) {
+                head = node;
+            } else {
+                element.prev.next = node;
+            }
+            element.prev = node;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> element = findElement(index);
+        return element.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> element = findElement(index);
+        T item = element.item;
+        element.item = value;
+        return item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> element = findElement(index);
+        unlink(element);
+        return element.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> element = head;
+        for (int i = 0; i < size; i++) {
+            if ((object == element.item) || (object != null && object.equals(element.item))) {
+                unlink(element);
+                return true;
+            }
+            element = element.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
```

Comment:
I'm not sure, but it seems to me when you add 2 "return" statements, you will never go through the second return  because the method will finish its work with the first one "return"
correct me if I'm wrong please